app/backend/hiStoryBackend/api/views/auth/views_v1.py

```diff
@@ -1,16 +1,48 @@
 from django.http import JsonResponse
 from api.helpers import json_response_helper as jrh
-
+from api.helpers import custom_helper
+from django.contrib.auth.hashers import check_password, make_password
+import json
+import jwt
+from api.models import User
+from django.core.exceptions import ValidationError
 
 def sign_in(request):
     if not request.method == 'POST':
         return jrh.bad_request(['Unsupported HTTP verb'])
 
     return JsonResponse({'operation': 'sign-in'})
 
-
 def sign_up(request):
     if not request.method == 'POST':
         return jrh.bad_request(['Unsupported HTTP verb'])
 
-    return JsonResponse({'operation': 'sign-up'})
+    r = json.loads(request.body)
+
+    username = r['username']
+    email = r['email']
+    password = r['password']
```

Comment:
Retrieving items like `r[key]` throws `KeyError` if that `key` is missing. Instead, you can use `r.get(key)`, which returns `None` if the key is missing.